### PR TITLE
delete session security fix

### PR DESCRIPTION
**Pull request by JerrySentry**
fixes: https://github.com/codecov/internal-issues/issues/469

Only allow sessions to be deleted if they are belonging to the authenticated user.

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. In 2022 this entity acquired Codecov and as result Sentry is going to need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: codecov_auth/commands/owner/interactors/delete_session.py [hunk 1]
```diff
@@ -7,14 +7,24 @@
...
 
 class DeleteSessionInteractor(BaseInteractor):
-    def validate(self):
+    def validate(self) -> None:
         if not self.current_user.is_authenticated:
             raise Unauthenticated()
 
     @sync_to_async
-    def execute(self, sessionid: int):
+    def execute(self, sessionid: int) -> None:
```
Comment (ajay-sentry) [inline codecov_auth/commands/owner/interactors/delete_session.py:15]: I thought there was a mypy rule that makes "return as none" implicit if no type is presented?

I could see it either way tho, related thread here: https://github.com/python/mypy/issues/7511

Comment (ajay-sentry) [reply codecov_auth/commands/owner/interactors/delete_session.py:15]: Yeah fair enough, Im good to keep as is

File: codecov_auth/tests/unit/views/test_okta.py [hunk 2]
```diff
@@ -10,8 +10,9 @@
 
 from codecov_auth.models import OktaUser
 from codecov_auth.tests.factories import OktaUserFactory, OwnerFactory, UserFactory
-from codecov_auth.views.okta import OktaLoginView, validate_id_token
+from codecov_auth.views.okta import OktaLoginView
 from codecov_auth.views.okta import auth as okta_basic_auth
+from codecov_auth.views.okta import validate_id_token
```
Comment (ajay-sentry) [inline codecov_auth/tests/unit/views/test_okta.py:15]: how come we split this to a separate line?

**Review by ajay-sentry — APPROVED** [review]

File: codecov_auth/tests/unit/views/test_okta.py (hunk 2 above)
Comment (JerrySentry) [reply codecov_auth/tests/unit/views/test_okta.py:15]: Uh ruff did that..

**Pull request closed without merging**